jazz_deployments-event-handler/test/test.js

```diff
@@ -1,9 +1,736 @@
+        }
+      }
+    }
+    index.getAuthResponse(result).then((auth) => {
+      expect(auth).to.eq(result.body.data.token);
+    })
+  })
+  it("should give error message when authentication fails ", () => {
+    let result = {
+      statusCode: 401,
+      body: {}
+    }
+    index.getAuthResponse(result).catch((err) => {
+      expect(err.message).to.eq('Invalid token response from API');
+    })
+  })
+
+})
+describe("checkforInterestedEvents", () => {
+  it("should return object with paramenter interested_event set to true", () => {
+    var record = event.Records[0];
+    var sequenceNumber = record.kinesis.sequenceNumber;
+    var encodedPayload = record.kinesis.data;
+    index.checkForInterestedEvents(encodedPayload, sequenceNumber, configData).then((res) => {
+      assert.isTrue(res.interested_event);
+    })
+  })
+  it("should reject with paramenter interested_event set to false", () => {
+    var payload = {
+      Item: {
+        EVENT_ID: {
+          S: '084f8c38-a01b-4ac9-943e-365f5de8ebe4'
+        },
+        TIMESTAMP: {
+          S: '2018-05-16T12:12:42:821'
+        },
+        REQUEST_ID: {
+          NULL: true
+        },
+        EVENT_HANDLER: {
+          S: 'JENKINS'
+        },
+        EVENT_NAME: {
+          S: 'CREATE_DEPLOYMENT'
+        },
+        SERVICE_NAME: {
+          S: 'test-02'
+        },
+        SERVICE_ID: {
+          S: '00001-test-serivice-id-00001'
+        },
+        EVENT_STATUS: {
+          S: 'STARTED'
+        },
+        EVENT_TYPE: {
+          S: 'NOT_SERVICE_DEPLOYMENT'
+        },
+        USERNAME: {
+          S: 'temp@testing.com'
+        },
+        EVENT_TIMESTAMP: {
+          S: '2018-05-16T12:12:41:083'
+        },
+        SERVICE_CONTEXT: {
+          S: '{"service_type":"api","branch":"","runtime":"nodejs","domain":"jazztest","iam_role":"arn:aws:iam::12345678:role/gitlabtest10001_test01","environment":"","region":"us-east-1","message":"input validation starts","created_by":"temp@testing.com"}'
+        }
+      }
+    }
+    var encoded = Buffer.from(JSON.stringify(payload)).toString('base64');
+    var sequenceNumber = "test_sequence01";
+    var encodedPayload = encoded;
+    index.checkForInterestedEvents(encodedPayload, sequenceNumber, configData).then((res) => {
+      assert.isFalse(res.interested_event);
+    })
+  })
+})
+describe("processEventRecords", () => {
+  var  procesEverntRecordStub
+  beforeEach(() => {
+    
+  })
+  afterEach(() => {
+    if (procesEverntRecordStub) {
+      procesEverntRecordStub.restore();
+    }
+  })
+  it("should resolve all for success scenario from processEventRecord",()=>{
+    procesEverntRecordStub = sinon.stub(index, "processEventRecord").resolves({
+      "status":"succesfully processed Event Rcord"
+    })
+    index.processEventRecords(event,configData,"temp_auth").then((obj)=>{
+      for(let i=0;i<obj.length;i++){expect(obj[i].status).to.eq("succesfully processed Event Rcord")}
```

Comment:
put this in separate lines and also do sinon.assert.calledOnce on stubs everywhere